error-handler.js

```diff
@@ -320,6 +331,81 @@ createHandler = function createHandler(options) {
   }
 };
 
+/**
+ * Maintenance middleware
+ * 
+ * @param {object} [options]
+ *
+ * @param {function} [options.status] - Returns Boolean to indicate
+ * the application is in maintenance mode. Default reads 
+ * ERR_HANDLER_MAINT_ENABLED environment variable - Value must be
+ * 'TRUE' to indicate an active maintenance condition, any other value 
+ * is false.
+ *
+ * @param {function} [options.retryAfter] - Returns
+ * Number or String value for Retry-After response header.
+ * Default reads ERR_HANDLER_MAINT_RETRYAFTER environment variable -
+ * Value must be a positive integer for retry after relative seconds,
+ * or an HTTP-date in GMT to indicate an absolte retry after period.
+ *
+ * @see http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.5.4
+ * @see http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.37
+ */
+createHandler.maintenance = function maintenance(options) {
+  options = options || {};
+
+  var
+    status = typeof options.status === 'function' ?
+      options.status :
+      function status () {
+        return ('' + process.env.ERR_HANDLER_MAINT_ENABLED)
+          .trim().toLowerCase() === 'true';
+      },
+
+    retryAfter = typeof options.retryAfter === 'function' ?
+      options.retryAfter :
+      function retryAfter() {
+        var result,
+            value = ('' + process.env.ERR_HANDLER_MAINT_RETRYAFTER).trim();
+        // fallback b/c user wants maintenance & no Retry-After means 500
+        var fallback = 3600;
+        // First, try seconds
+        result = parseInt(value, 10);
```

Comment:
You set result 3 times instead of naming it different things in order to illustrate what you're doing in each line.

The line `result = parseInt(value, 10);` tells me nothing.

The line `envSetting = parseInt(value, 10);` tells me that you're getting the value of the environment variable.